### ISSUE TITLE
Can't use a name starting with an upper letter

### ISSUE THREAD

**Issue by sorawee** [ucsd-progsys/liquidhaskell#691]
A student in our class discovers the following bug:

```
{-@ id :: foo: Int -> {v : Int | v = foo } @-}
id :: Int -> Int
id x = x
```

is fine.

```
{-@ id :: FOO: Int -> Int @-}
id :: Int -> Int
id x = x
```

is also fine.

```
{-@ id :: FOO: Int -> {v : Int | v == FOO} @-}
id :: Int -> Int
id x = x
```

results in a GHC error (Not in scope: variable `FOO').

We should either disallow anyone from using a name starting with an upper letter (with descriptive error message), so that both 2) and 3) result in an error, or fix this so that it's not an error...

**Comment by ranjitjhala** [comment]
Oops. I am surprised the second thing was accepted. Uppercase should not be allowed for binders!

**Comment by ranjitjhala** [comment]
This line

https://github.com/ucsd-progsys/liquidhaskell/blob/develop/src/Language/Haskell/Liquid/Parse.hs#L397

should be modified so we use `lowerIdP` instead of `binderP` ... (assuming it doesn't break anything...)

**Comment by ranjitjhala** [comment]
Fixed by 55e7fd6a949ef81f9757a994276fe6313c0a67c2 (oops looks like there's a conflict...)
